### PR TITLE
fix(config): move endpoint, platform to GlobalConfig

### LINKED ISSUE #15859: refactor: move platform and endpoint to GlobalConfig
### Describe the proposed change(s).

Per above

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Moves `platform` and `endpoint` to `GlobalConfig`.

## Context

Closes #15859

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/gitlabci-include/extract.ts [hunk 1]
```diff
@@ -44,10 +42,8 @@ export function extractPackageFile(
     for (const includeObj of includes) {
       if (includeObj?.file && includeObj.project) {
         const dep = extractDepFromIncludeFile(includeObj);
-        if (config.endpoint) {
-          dep.registryUrls = [
-            config.endpoint.replace(regEx(/\/api\/v4\/?/), ''),
-          ];
+        if (platform === 'gitlab' && endpoint) {
```
Comment (rarkins) [inline lib/modules/manager/gitlabci-include/extract.ts:45]: This is essentially a fix, not refactor, but I thought it OK to include

File: lib/modules/manager/npm/post-update/index.ts [hunk 2]
```diff
@@ -269,7 +269,7 @@ export async function writeUpdatedPackageFiles(
     const massagedFile = JSON.parse(packageFile.contents.toString());
     try {
       const { token } = hostRules.find({
-        hostType: config.platform,
+        hostType: 'github',
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/index.ts:272]: This is also a fix

File: lib/modules/manager/npm/post-update/index.ts [hunk 3]
```diff
@@ -507,7 +507,7 @@ export async function getAdditionalFiles(
   let token: string | undefined;
   try {
     ({ token } = hostRules.find({
-      hostType: config.platform,
+      hostType: 'github',
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/index.ts:510]: Also a fix

File: lib/modules/manager/gitlabci-include/extract.ts [hunk 4]
```diff
@@ -24,12 +25,9 @@ function extractDepFromIncludeFile(includeObj: {
   return dep;
 }
 
-export function extractPackageFile(
-  content: string,
-  _packageFile: string,
-  config: ExtractConfig
-): PackageFile | null {
+export function extractPackageFile(content: string): PackageFile | null {
```
Comment (viceice) [inline lib/modules/manager/gitlabci-include/extract.ts:28]: shouldn't this function match the manager interface? even there are unused parameters? 
just for API consistency.

File: lib/workers/global/index.ts [hunk 5]
```diff
@@ -31,9 +32,10 @@ export async function getRepositoryConfig(
     globalConfig,
     is.string(repository) ? { repository } : repository
   );
+  const { platform } = GlobalConfig.get();
```
Comment (viceice) [inline lib/workers/global/index.ts:35]:
```suggestion
  const platform = GlobalConfig.get('platform');
```
is probably better for null safety check.

File: lib/workers/repository/init/index.ts [hunk 6]
```diff
@@ -17,8 +18,9 @@ function initializeConfig(config: RenovateConfig): RenovateConfig {
 
 function warnOnUnsupportedOptions(config: RenovateConfig): void {
   if (config.filterUnavailableUsers && !platform.filterUnavailableUsers) {
+    const { platform } = GlobalConfig.get();
```
Comment (viceice) [inline lib/workers/repository/init/index.ts:21]:
```suggestion
    const platform = GlobalConfig.get('platform');
```

File: lib/workers/repository/onboarding/branch/config.ts [hunk 7]
```diff
@@ -37,13 +38,13 @@ async function getOnboardingConfig(
 
   if (!orgPreset) {
     // Check for org/.{{platform}}
+    const { platform } = GlobalConfig.get();
```
Comment (viceice) [inline lib/workers/repository/onboarding/branch/config.ts:41]:
```suggestion
    const platform = GlobalConfig.get('platform');
```

**Review by viceice — COMMENTED** [review]
only some small suggestions

File: lib/modules/manager/gitlabci-include/extract.ts (hunk 4 above)
Comment (rarkins) [reply lib/modules/manager/gitlabci-include/extract.ts:28]: I'm not sure if I misunderstand, but don't we do the same as this for many/most managers already? e.g.

image

Comment (viceice) [reply lib/modules/manager/gitlabci-include/extract.ts:28]: ok, 🤔

**Review by viceice — APPROVED** [review]
nice, makes preset handling more simple

**Pull request merged**